src/main/steps/c100-rebuild/uploadDocumentController.ts

```diff
@@ -0,0 +1,224 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types */
+import autobind from 'autobind-decorator';
+import { Response } from 'express';
+import FormData from 'form-data';
+import { isNull } from 'lodash';
+
+import { DocumentUploadResponse } from '../../app/case/C100CaseApi';
+import { C100DocumentInfo } from '../../app/case/definition';
+import { AppRequest } from '../../app/controller/AppRequest';
+import { AnyObject, PostController } from '../../app/controller/PostController';
+import { FormFields, FormFieldsFn } from '../../app/form/Form';
+import { isFileSizeGreaterThanMaxAllowed, isValidFileFormat } from '../../app/form/validation';
+import { C100_CONSENT_ORDER_UPLOAD, C100_MIAM_UPLOAD } from '../urls';
+
+/* The UploadDocumentController class extends the PostController class and overrides the
+PostDocumentUploader method */
+@autobind
+export default class UploadDocumentController {
+  private parent;
+  constructor(protected readonly fields: FormFields | FormFieldsFn) {
+    this.parent = new PostController(fields);
+  }
+  /**
+   * A recursive function that calls itself.
+   * @param req - AppRequest<AnyObject>
+   * @param {Response} res - Response - This is the response object txhat will be sent back to the client.
+   */
+
+  public async post(req: AppRequest<AnyObject>, res: Response): Promise<void> {
+    const { files }: AppRequest<AnyObject> = req;
+    req.session.errors = [];
```

Comment:
Why are we initialising errors as empt in so many places ?
line num - 32, 54, 123, 147